src/common/data.py

```diff
@@ -51,3 +51,105 @@ def generate(self):
         y = np.squeeze(y)
 
         return X, y
+
+
+class ClassificationDataGenerator():
+    """Generator for regression data"""
+    def __init__(self,
+                 n_label_classes: int,
+                 batch_size: int,
+                 n_features: int,
+                 n_informative: int,
+                 n_targets: int,
+                 bias: float,
+                 noise: float,
+                 seed: int):
+        """Initializes the generator"""
+        self.n_label_classes = n_label_classes
+        self.batch_size = batch_size
+        self.n_features = n_features
+        self.n_informative = min(n_features, n_informative)
+        self.n_targets = n_targets
+        self.bias = bias
+        self.noise = noise
+        self.generator = np.random.RandomState(seed)
+
+        # Generate a ground truth model with only n_informative features being non-zeros
+        self.ground_truth = np.zeros((self.n_features, self.n_targets))
+        self.ground_truth[:n_informative, :] = 100 * self.generator.rand(self.n_informative, self.n_targets)
+
+    def generate(self):
+        """Generate one batch of data.
+
+        Returns:
+            X (numpy.ndarray)
+            y (numpy.ndarray)
+        """
+        # Randomly generate a well conditioned input set
+        X = self.generator.randn(self.batch_size, self.n_features)
+        y = np.dot(X, self.ground_truth) + self.bias
+
+        # Add noise
+        if self.noise > 0.0:
+            y += self.generator.normal(scale=self.noise, size=y.shape)
+
+        # create n_label_classes ranking labels
+        y = ((y - min(y))/(max(y)-min(y))*self.n_label_classes).astype(int)
```

Comment:
how does that work with n_targets? it looks like n_targets is not being used in any way, but I'm curious if that should be the case here so that each class corresponds to a different regression model?

if you don't want to use that approach, then the code should not use n_targets at all?

Reply:
I removed n_targets altogether, we can bring it back if we want to create more complicated data generation later

Reply:
I removed all the n_targets, we can bring them back if needed in future